internal/remote_actions.go

```diff
@@ -0,0 +1,75 @@
+//go:build !disable_remoteactions
+
+package internal
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/foundriesio/fioconfig/sotatoml"
+)
+
+type raInitCallback struct {
+	newActions []byte
+}
+
+func (r *raInitCallback) ConfigFiles(app *App) []ConfigFileReq {
+	actionsDir := "/usr/share/fioconfig/actions"
+	var actions []string
+	entries, err := os.ReadDir(actionsDir)
+	if err != nil && errors.Is(err, os.ErrNotExist) {
+		// okay - treat as disabled
+		actions = nil
+	} else if err != nil {
+		// This is a bug - we can't read the directory for some reason
+		log.Printf("Unable to initialize remote actions: %s", err)
+		return nil
+	} else {
+		for _, entry := range entries {
+			actions = append(actions, entry.Name())
+		}
+	}
+	actionsStr := strings.Join(actions, ",")
+
+	content, err := os.ReadFile(filepath.Join(app.SecretsDir, "remote-actions"))
+	if err != nil {
+		if os.IsNotExist(err) {
+			content = nil
+		} else {
+			log.Printf("Unable to read configured remote actions: %s", err)
+			return nil
+		}
+	}
+
+	if !bytes.Equal(content, []byte(actionsStr)) {
+		log.Printf("Configured remote actions changed from %s->%s", string(content), actionsStr)
+		file := ConfigFileReq{
+			Name:        "fio-remote-actions",
+			Unencrypted: true,
+			Value:       actionsStr,
+		}
+		r.newActions = []byte(actionsStr)
+		return []ConfigFileReq{file}
+	}
+
+	// prevent init logic from calling OnComplete by removing ourselves
+	delete(initCallbacks, "remote-actions")
+	return nil
+}
+
+func (r raInitCallback) OnComplete(app *App) {
+	triggerFile := filepath.Join(app.SecretsDir, "remote-actions")
+	if err := sotatoml.SafeWrite(triggerFile, r.newActions); err != nil {
+		log.Printf("Unable to save current remote actions: %s", err)
+	} else {
+		delete(initCallbacks, "remote-actions")
```

Comment:
Maybe, declare a constant for the file name?

Reply:
This isn't a file name - its the name of the callback. My force push has cleaned some things up around this.